### PR TITLE
feat(auth-server): add more backend error data

### PR DESCRIPTION
**Pull request by bbangert**
Because:

* EndpointError has no stack trace, and no information about what
  request was inflight, to what service, with what exact response.

This commit:

* Determines if the error object has EndpointError attributes and if
  so extracts additional data about the underlying backend request
  call, method, path, and full response error data.

Here's what an EndpointError with this PR includes in Sentry:
image

### CODE REVIEW

File: packages/fxa-auth-server/lib/sentry.js [hunk 1]
```diff
@@ -88,6 +88,9 @@ async function configureSentry(server, config) {
       beforeSend(event, hint) {
         return filterSentryEvent(event, hint);
       },
+      integrations: [
+        new Sentry.Integrations.LinkedErrors({ key: 'jse_cause' }),
```
Comment (bbangert) [inline packages/fxa-auth-server/lib/sentry.js]: I hadn't noticed before that the default Sentry LinkedErrors integration looks for `cause`. This updates it to look for the proper cause key for `verror`.

File: packages/fxa-auth-server/lib/sentry.js [hunk 2]
```diff
@@ -131,6 +129,33 @@ async function configureSentry(server, config) {
             return sentryEvent;
           });
           scope.setExtra('exception', exception);
+          const cause = verror.cause(err);
```
Comment (shane-tomlinson) [inline packages/fxa-auth-server/lib/sentry.js]: Hmm, I find it somewhat distressing that the unit tests for the sentry module are somewhat non-existent: https://github.com/mozilla/fxa/blob/d8b7629291ec6e0c3846c6c8954521cffd145884/packages/fxa-auth-server/test/local/sentry.js#L7

**Comment by bbangert** [comment]
@shane-tomlinson okey, a test has been added that verifies the error and nested error is extracted ok.

File: packages/fxa-auth-server/test/local/sentry.js [hunk 3]
```diff
@@ -5,30 +5,81 @@
...
+      statusCode: 500,
+    };
+    const sentryCaptureSpy = sinon.stub(Sentry, 'captureException');
+    const scopeContextSpy = sinon.fake();
+    const scopeSpy = {
+      addEventProcessor: sinon.fake(),
+      setContext: scopeContextSpy,
+      setExtra: sinon.fake(),
+    };
+    sinon.replace(Sentry, 'withScope', fn => fn(scopeSpy));
```
Comment (shane-tomlinson) [inline packages/fxa-auth-server/test/local/sentry.js]: sinon's fakes, TIL! https://sinonjs.org/releases/latest/fakes/

**Pull request merged**